overlord/snapstate/component.go

```diff
@@ -37,9 +38,19 @@ import (
 // InstallComponents installs all of the components in the given names list. The
 // snap represented by info must already be installed, and all of the components
 // in names should not be installed prior to calling this function.
-//
-// TODO:COMPS: respect the transaction that is passed to this function
 func InstallComponents(ctx context.Context, st *state.State, names []string, info *snap.Info, opts Options) ([]*state.TaskSet, error) {
+	if opts.Flags.Transaction != client.TransactionAllSnaps && opts.Flags.Lane != 0 {
+		return nil, errors.New("cannot specify a lane without setting transaction to \"all-snaps\"")
+	}
+
+	if opts.Flags.Transaction == client.TransactionAllSnaps && opts.Flags.Lane == 0 {
+		opts.Flags.Lane = st.NewLane()
```

Comment:
this code is not tested, also we have already some copies of it, maybe it should be a method opts?